### PR TITLE
Resist time lowered, clean and refactor.

### PR DESCRIPTION
**Pull request by Rohesie**
Diminished the nest resist timer to 60 seconds, down from 90.
This needed to be done due to the different laval growth accelerants plus the nature of halloss damage, which makes the victim unable to resist for a long time after being buckled.

Cleaned and refactored the xeno nest code to diminish the presence of evil a little, incidentally.

## Changelog
:cl:
balance: Xeno nest resist time lowered to 60 seconds down from 90.
/:cl:

### CODE REVIEW

**Comment by Ketrai** [comment]
Rohesie, if you make this change it's going to suck a lot for the hive to capture people with a mag harnassed weapon that's good in CQC. Shotgun, minigun, ZX.

People will just start killing captures against instead of bothering.

**Comment by Rohesie** [comment]
A full minute is a long time in the game. Could always buff the effect of the nest while reducing the larval growth one. The key is finding a proper balance.
Right now capturing is not really hard at all. My gameplay as a xeno tells me so.

File: code/game/objects/structures/stool_bed_chair_nest/xeno_nest.dm [hunk 1]
```diff
@@ -10,153 +12,140 @@
...
+/obj/structure/bed/nest/attackby(obj/item/I, mob/living/user)
+	if(istype(I, /obj/item/grab))
+		var/obj/item/grab/G = I
+		if(!isliving(G.grabbed_thing))
+			return
+		var/mob/living/L = G.grabbed_thing
+		to_chat(user, "<span class='notice'>You place [L] on [src].</span>")
+		L.forceMove(loc)
 		return TRUE
 	else
```
Comment (spookydonut) [inline code/game/objects/structures/stool_bed_chair_nest/xeno_nest.dm]: you dont need this else

File: code/game/objects/structures/stool_bed_chair_nest/xeno_nest.dm [hunk 2]
```diff
@@ -10,153 +12,140 @@
...
+			return
+		if(resisting_time + NEST_RESIST_TIME > world.time)
+			to_chat(buckled_mob, "<span class='warning'>You're already trying to free yourself. Give it some time.</span>")
+			return
+		buckled_mob.visible_message("<span class='danger'>\The [buckled_mob] breaks free from \the [src]!</span>",\
+		"<span class='danger'>You pull yourself free from \the [src]!</span>",\
+		"<span class='notice'>You hear squelching.</span>")
+		unbuckle()
+
+	add_fingerprint(user)
```
Comment (spookydonut) [inline code/game/objects/structures/stool_bed_chair_nest/xeno_nest.dm]: put this at the start

File: code/game/objects/structures/stool_bed_chair_nest/xeno_nest.dm [hunk 3]
```diff
@@ -10,153 +12,140 @@
...
+			return
+		buckled_mob.visible_message("<span class='notice'>\The [user] pulls \the [buckled_mob] free from \the [src]!</span>",\
+		"<span class='notice'>\The [user] pulls you free from \the [src].</span>",\
+		"<span class='notice'>You hear squelching.</span>")
+		playsound(loc, "alien_resin_move", 50)
+		if(ishuman(buckled_mob))
+			var/mob/living/carbon/human/H = buckled_mob
+			H.last_unbuckled = world.time
+		unbuckle()
+	else
```
Comment (spookydonut) [inline code/game/objects/structures/stool_bed_chair_nest/xeno_nest.dm]: dont need this else

File: code/game/objects/structures/stool_bed_chair_nest/xeno_nest.dm [hunk 4]
```diff
@@ -10,153 +12,140 @@
...
+		buckled_mob.visible_message("<span class='danger'>\The [buckled_mob] breaks free from \the [src]!</span>",\
+		"<span class='danger'>You pull yourself free from \the [src]!</span>",\
+		"<span class='notice'>You hear squelching.</span>")
+		unbuckle()
+
+	add_fingerprint(user)
+
+
+/obj/structure/bed/nest/proc/unbuckle_time_message(mob/living/user)
+	if(QDELETED(src) || QDELETED(user) || user != buckled_mob)
```
Comment (spookydonut) [inline code/game/objects/structures/stool_bed_chair_nest/xeno_nest.dm]: if src is deleted the timers will be too.

File: code/game/objects/structures/stool_bed_chair_nest/xeno_nest.dm (hunk 2 above)
Comment (Rohesie) [reply code/game/objects/structures/stool_bed_chair_nest/xeno_nest.dm]: That will generate some quirky behavior, but sure.

File: code/game/objects/structures/stool_bed_chair_nest/xeno_nest.dm [hunk 5]
```diff
@@ -10,153 +12,140 @@
...
+		if(user.incapacitated())
+			return
+		buckled_mob.visible_message("<span class='notice'>\The [user] pulls \the [buckled_mob] free from \the [src]!</span>",\
+		"<span class='notice'>\The [user] pulls you free from \the [src].</span>",\
+		"<span class='notice'>You hear squelching.</span>")
+		playsound(loc, "alien_resin_move", 50)
+		if(ishuman(buckled_mob))
+			var/mob/living/carbon/human/H = buckled_mob
+			H.last_unbuckled = world.time
+		unbuckle()
```
Comment (spookydonut) [inline code/game/objects/structures/stool_bed_chair_nest/xeno_nest.dm]: Need a return after this

Comment (Rohesie) [reply code/game/objects/structures/stool_bed_chair_nest/xeno_nest.dm]: Missed that one, good catch.

**Pull request merged**